diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -411,67 +373,68 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
+    private fun searchDotOrSafeAccess(node: ASTNode, dotList: MutableList<ASTNode>) {
+        if (node.elementType == DOT_QUALIFIED_EXPRESSION || node.elementType == SAFE_ACCESS_EXPRESSION || node.elementType == POSTFIX_EXPRESSION) {
+            node.getChildren(null)
+                .forEach {
+                    searchDotOrSafeAccess(it, dotList)
+                }
+            if (node.elementType != POSTFIX_EXPRESSION) {
+                dotList.add(node)
+            }
         }
-        returnList.add(expression)
-        return returnList
     }
 
     /**
-     * Runs through the sorted list [rightBinList], finds its last element, the type of which is included in the set [typesList] and adds it in the list [returnList]
+     * Finds the first binary expression closer to the separator
      */
-    @Suppress("TYPE_ALIAS", "UnsafeCallOnNullableType")
-    private fun addInSmartListBinExpression(
-        returnList: MutableList<Pair<ASTNode, Int>?>,
-        rightBinList: List<Pair<ASTNode, Int>>,
-        typesList: List<IElementType>,
-        configuration: LineLengthConfiguration
-    ) {
-        val expression = rightBinList.firstOrNull { (it, offset) ->
-            val binOperationReference = it.getFirstChildWithType(OPERATION_REFERENCE)
-            offset + (it.getFirstChildWithType(OPERATION_REFERENCE)?.text!!.length ?: 0) <= configuration.lineLength + 1 &&
-                binOperationReference!!.firstChildNode.elementType in typesList
-        }
-        returnList.add(expression)
+    @Suppress("UnsafeCallOnNullableType")
+    private fun searchRightSplitAfterOperationReference(
+        parent: ASTNode,
+        configuration: LineLengthConfiguration,
+    ): Pair<ASTNode, Int>? {
+        val list: MutableList<ASTNode> = mutableListOf()
+        searchBinaryExpression(parent, list)
+        return list.asSequence()
+            .map {
+                it to positionByOffset(it.getFirstChildWithType(OPERATION_REFERENCE)!!.startOffset).second
+            }
+            .sortedBy { it.second }
+            .lastOrNull { (it, offset) ->
+                offset + (it.getFirstChildWithType(OPERATION_REFERENCE)?.text?.length ?: 0) <= configuration.lineLength + 1
+            }
     }
 
     /**
-     * Finds the first binary expression closer to the separator
+     * Finds the first dot or safe access closer to the separator
      */
-    @Suppress("UnsafeCallOnNullableType")
-    private fun searchRightSplitInBinaryExpression(parent: ASTNode, configuration: LineLengthConfiguration): Pair<ASTNode, Int>? {
-        val binList: MutableList<ASTNode> = mutableListOf()
-        searchBinaryExpression(parent, binList)
-        return binList.map {
-            it to positionByOffset(it.getFirstChildWithType(OPERATION_REFERENCE)!!.startOffset).second
-        }
+    @Suppress("MAGIC_NUMBER", "MagicNumber")
```

Comment:
@akuleshov7 @petertrr, as I got, there are two different rules from diktat and detect -- is it possible to support suppressing from detect on diktat level to avoid duplication?